doc/release/2.8.1.rst

```diff
@@ -0,0 +1,347 @@
+
+Compatibility
+-------------
+
+Tarantool 2.x is backward compatible with Tarantool 1.10.x in the binary
+data layout, client-server protocol, and replication protocol.
+
+Please
+`upgrade <https://www.tarantool.io/en/doc/latest/book/admin/upgrades/>`__
+using the ``box.schema.upgrade()`` procedure to unlock all the new
+features of the 2.x series.
+
+
+Functionality added or changed
+------------------------------
+
+Core
+~~~~
+
+-   The ``exclude_null`` option can now be used in the index part definition.
+    With this option, the index filters and doesn't store tuples with "null" value
+    of the corresponding part part (:tarantool-issue:`4480`).
+
+    For example, an index created with
+    ``s:create_index('sk', {parts={{2, 'number', exclude_null=true}}})``
+    will ignore tuples ``{1, null}`` and ``{2, null}``,
+    but will not ignore ``{null, 1}`` or ``{1, 1}``.
+
+-   Added a ``slab_alloc_granularity`` option to ``box.cfg{}``. This option allows
+    user to set multiplicity of memory allocation in a small allocator.
+    The value of ``slab_alloc_granularity`` must be exponent of two and >= 4
+    (:tarantool-issue:`5518`).
+
+-   Previously, Lua `on_shutdown` triggers started sequentially.
+    Now each trigger starts in a separate fiber.
+    By default, Tarantool waits for the triggers to finish for 3.0 seconds.
+    User has the option to change this value using the new
+    ``box.ctl.set_on_shutdown_timeout`` function.
+
+    When the timeout expires, Tarantool stops immediately
+    without waiting for the other triggers to finish.
+
+-   Tarantool module developers can now register functions to call when Tarantool stops
+    with the new ``on_shutdown`` API, (:tarantool-issue:`5723`).
+
+-   Introduced the concept of WAL queue and the new configuration option
+    ``wal_queue_max_size``, measured in bytes.
+    The default value is 16 Mb.
+    The option helps limit the pace at which replica submits new transactions to the WAL.
+    The limit is checked every time a transaction from the master is
+    submitted to the replica’s WAL.
+    The space taken by the transaction is considered empty once it’s successfully written
+    (:tarantool-issue:`5536`).
+
+-   The information about the state of synchronous replication is now available via the
+    ``box.info.synchro`` interface (:tarantool-issue:`5191`).
+
+-   Tarantool can now :ref:`run multiple iproto threads <cfg_networking-iproto_threads>`.
+    It is useful in some specific workloads where iproto thread is the bottleneck of
+    throughput (:tarantool-issue:`5645`).
+
+-   Update operations can’t insert with gaps. This patch changes the
+    behavior so that the update operation fills the missing fields with
+    nulls (:tarantool-issue:`3378`).
+
+-   The new ``box.lib`` module allows loading and executing C
+    stored procedures on read-only nodes (:tarantool-issue:`4642`).
+
+-   Configuration options in `box.cfg` can now be defined with environment variables
```

Comment:
box.cfg in notable changes and in the list below is wrapped into a span tag rather than italic.